src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -141,27 +150,50 @@ export class SharedCredentialsProvider implements CredentialsProvider {
         return undefined
     }
 
-    public async getCredentials(): Promise<AWS.Credentials> {
-        // Profiles with references involving non-aws partitions need help getting the right STS endpoint
-        // when resolving SharedIniFileCredentials. We set the global sts configuration with a suitable region
-        // only to perform the resolve, then reset it.
-        // This hack can be removed when https://github.com/aws/aws-sdk-js/issues/3088 is addressed.
-        const originalStsConfiguration = AWS.config.sts
-
-        try {
-            // Profiles with references involving non-aws partitions need help getting the right STS endpoint
-            this.applyProfileRegionToGlobalStsConfig()
-            //  SSO entry point
-            if (this.isSsoProfile()) {
-                const ssoCredentialProvider = this.makeSsoProvider()
-                return await ssoCredentialProvider.refreshCredentials()
+    // Patches 'source_profile' credentials as static representations, which the SDK can handle in all cases
```

Comment:
i think an extended comment is justified here. Just pasting your comment from this PR would be good:
```suggestion
    // Patches 'source_profile' credentials as static representations, which the SDK can handle in all cases.
    //
    // The SDK is unable to resolve source_profile fields when the source profile uses SSO/MFA/credential_process.
    // We can handle this resolution ourselves, giving the SDK the resolved credentials by 'pre-loading' them.
```